src/tools/tidy/src/ui_tests.rs

```diff
@@ -66,7 +93,19 @@ pub fn check(path: &Path, bad: &mut bool) {
     let paths = [ui.as_path(), ui_fulldeps.as_path()];
     crate::walk::walk_no_read(&paths, |_, _| false, &mut |entry| {
         let file_path = entry.path();
-        if let Some(ext) = file_path.extension() {
+        if let Some(ext) = file_path.extension().and_then(OsStr::to_str) {
+            // check for files that don't have an expected extension, they're probably typos or not meant to be here
+            if !EXPECTED_TEST_FILE_EXTENSIONS.contains(&ext) {
+                // note: checking that "not any" of the paths match is checking if none match
+                if !EXTENSION_EXCEPTION_PATHS.iter().any(|path| file_path.ends_with(path)) {
+                    tidy_error!(
+                        bad,
+                        "file {} has unexpected extension {}",
+                        file_path.display(),
+                        ext
+                    );
+                }
+            }
```

Comment:
```suggestion
            // check for files that don't have an expected extension, they're probably typos or not meant to be here
            if !EXPECTED_TEST_FILE_EXTENSIONS.contains(&ext)
                // note: checking that "not any" of the paths match is checking if none match
                && !EXTENSION_EXCEPTION_PATHS.iter().any(|path| file_path.ends_with(path))
            {
                tidy_error!(
                    bad,
                    "file {} has unexpected extension {ext}",
                    file_path.display(),
                );
            }
```

Reply:
Btw alternatively the second check can be written as `.all(|| !ends_with)` which is probably more intuitive